
On OmniOS, several libraries have had their functionality moved into libc
but the old libraries persist as stubs for compatibility reasons. Two such
libraries are libdl.so & libpthread.so

There's no specific problem with an application or library linking with these
(just a slight overhead) but when a library is used by a component within
illumos-omnios, a check is made which generates a warning if an
unecessary library is linked.

Openssl is one of the libraries used by illumos-omnios and therefore we
patch the configuration script to stop linking against these two libraries.

We also patch out the desire to omit the frame pointer and add in dwarf-2
debug information which can be converted to CTF.

diff -wpruN '--exclude=*.orig' a~/Configurations/10-main.conf a/Configurations/10-main.conf
--- a~/Configurations/10-main.conf	1970-01-01 00:00:00
+++ a/Configurations/10-main.conf	1970-01-01 00:00:00
@@ -208,7 +208,7 @@ my %targets = (
         inherit_from     => [ "BASE_unix" ],
         template         => 1,
         lib_cppflags     => "-DFILIO_H",
-        ex_libs          => add("-lsocket -lnsl -ldl"),
+        ex_libs          => add("-lsocket -lnsl"),
         dso_scheme       => "dlfcn",
         thread_scheme    => "pthreads",
         shared_target    => "self",
@@ -226,7 +226,7 @@ my %targets = (
         CC               => "gcc",
         CFLAGS           => add_before(picker(default => "-Wall",
                                               debug   => "-O0 -g",
-                                              release => "-O3 -fomit-frame-pointer")),
+                                              release => "-O3 -gdwarf-2")),
         cflags           => add(threads("-pthread")),
         lib_cppflags     => add("-DL_ENDIAN"),
         ex_libs          => add(threads("-pthread")),
@@ -247,7 +247,7 @@ my %targets = (
         CC               => "gcc",
         CFLAGS           => add_before(picker(default => "-Wall",
                                               debug   => "-O0 -g",
-                                              release => "-O3")),
+                                              release => "-O3 -gdwarf-2")),
         cflags           => add_before("-m64", threads("-pthread")),
         lib_cppflags     => add("-DL_ENDIAN"),
         ex_libs          => add(threads("-pthread")),
